### PR TITLE
Add boilerplate

### PR DESCRIPTION
**Pull request by irinakhismatullina**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+dist: bionic
+
+language: go
+
+go:
+  - 1.12.x
+  - 1.13.x
+
+services:
+  - docker
```
Comment (vmarkovtsev) [inline .travis.yml]: We don't need docker yet

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+env:
+  global:
+    - GO111MODULE=on
+
+install:
+  - travis_retry make dependencies
+
```
Comment (vmarkovtsev) [inline .travis.yml:15]: Missing:

```yaml
script:
    - make check-style
    - make test
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Contributing Guidelines
+
+go-YouTokenToMe project is GPL 3.0 licensed and accepts
```
Comment (vmarkovtsev) [inline CONTRIBUTING.md]: Apache 2.0

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+contributions via GitHub pull requests.  This document outlines some of the
+conventions on development workflow, commit message formatting, contact points,
+and other resources to make it easier to get your contribution accepted.
+
+## How to Contribute
+
+Pull Requests (PRs) are the main and exclusive way to contribute to the official src-d/go-YouTokenToMe project.
+In order for a PR to be accepted it needs to pass a list of requirements:
+
+- All PRs must be written in idiomatic Go, formatted according to [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), and without any warnings from [go lint](https://github.com/golang/lint) nor [go vet](https://golang.org/cmd/vet/).
```
Comment (vmarkovtsev) [inline CONTRIBUTING.md]:
```suggestion
- All PRs must be written in idiomatic Go, formatted according to gofmt and [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), and without any warnings from [go lint](https://github.com/golang/lint) nor [go vet](https://golang.org/cmd/vet/).
```

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (vmarkovtsev) [inline LICENSE]: Apache 2.0

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+.POSIX:
+check-style:
+	golint -set_exit_status ./...
+	# Run `make fix-style` to fix style errors
+	test -z "$$(gofmt -s -d .)"
+	test -z "$$(goimports -d .)"
+	go vet
+
+install-dev-deps:
+	go get -v golang.org/x/lint/golint github.com/mjibson/esc golang.org/x/tools/cmd/goimports
```
Comment (vmarkovtsev) [inline Makefile:39]: golint should be installed too

File: README.md [hunk 7]
```diff
@@ -1 +1,3 @@
-# go-YouTokenToMe
\ No newline at end of file
+# go-YouTokenToMe
+
+go-YouTokenToMe is a Golang port of [YoutTokenToMe](https://github.com/VKCOM/YouTokenToMe) - a computationally efficient implementation of Byte Pair Encoding [[Sennrich et al.](https://www.aclweb.org/anthology/P16-1162/)].
```
Comment (vmarkovtsev) [inline README.md]:
```suggestion
go-YouTokenToMe is a Go port of [YoutTokenToMe](https://github.com/VKCOM/YouTokenToMe) - a computationally efficient implementation of Byte Pair Encoding [[Sennrich et al.](https://www.aclweb.org/anthology/P16-1162/)]. Only inference is supported, no training.
```

File: go.mod [hunk 8]
```diff
@@ -0,0 +1,3 @@
+module github.com/irinakhismatullina/go-YouTokenToMe
```
Comment (vmarkovtsev) [inline go.mod]:
```suggestion
module github.com/src-d/go-YouTokenToMe
```

**Review by vmarkovtsev — APPROVED** [review]

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+dist: bionic
```
Comment (vmarkovtsev) [inline .travis.yml:1]: Oops no approve

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: Makefile (hunk 6 above)
Comment (irinakhismatullina) [reply Makefile:39]: golint is already present

**Review by vmarkovtsev — APPROVED** [review]

File: .travis.yml [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+env:
+  global:
+    - GO111MODULE=on
+
+install:
+  - travis_retry make dependencies
+
+script:
+  - make test-coverage
```
Comment (vmarkovtsev) [inline .travis.yml]: Only one of three should be left

File: .travis.yml [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+ - style
+
+matrix:
+  fast_finish: true
+  include:
+    - stage: style
+      go: 1.13.x
+      addons:
+        apt:
+          packages:
```
Comment (vmarkovtsev) [inline .travis.yml]: We don't need any Python here. Remove it from everywhere here and Makefile

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: go.mod [hunk 12]
```diff
@@ -0,0 +1,10 @@
+module github.com/src-d/go-YouTokenToMe
+
+go 1.12
+
+require (
+	github.com/mjibson/esc v0.2.0 // indirect
```
Comment (vmarkovtsev) [inline go.mod]: The requires should be empty - we don have any code! Run `go mod tidy`

File: go.sum [hunk 13]
```diff
@@ -0,0 +1,16 @@
+github.com/mjibson/esc v0.2.0 h1:k96hdaR9Z+nMcnDwNrOvhdBqtjyMrbVyxLpsRCdP2mA=
```
Comment (vmarkovtsev) [inline go.sum]: Let's exclude `go.sum` from the repo - it is frequently updated and we don't care.

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+script:
+  - make test
+
+stages:
+ - style
+
+matrix:
+  fast_finish: true
+  include:
+    - stage: style
```
Comment (vmarkovtsev) [inline .travis.yml]: Let's ignore the stages and check the style before running the tests. It will be faster.

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

**Comment by irinakhismatullina** [comment]
@vmarkovtsev Still get `can't load package: package .: no Go files in /home/travis/gopath/src/github.com/src-d/go-YouTokenToMe` https://travis-ci.com/src-d/go-YouTokenToMe/jobs/245798527#L270

**Comment by irinakhismatullina** [comment]
And now `can't load package: package github.com/src-d/go-YouTokenToMe: unknown import path "github.com/src-d/go-YouTokenToMe": cannot find module providing package github.com/src-d/go-YouTokenToMe`

**Comment by irinakhismatullina** [comment]
Finally ready

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**